### ISSUE TITLE
Unused parameter raises ArgumentError

### ISSUE THREAD

**Issue by lungd** [SciML/ModelingToolkit.jl#364]
I get an `ERROR: ArgumentError: invalid index: nothing of type Nothing` with the following code:
```
using ModelingToolkit, OrdinaryDiffEq

@parameters t a b
@variables x(t)
@derivatives D'~t

equ = [
	D(x) ~ a,  # no error with D(x) ~ a*b
]

u0 = [
	x => 1.0,
]

p = [	
	a => 0.1,
	b => 0.2,
]

sys = ODESystem(equ)

tspan = (0.0, 1.0)
prob = ODEProblem(sys, u0, tspan, p, jac=true, sparse=true)
sol = solve(prob, Tsit5())
```

I am not sure if this is intended behavior but it took me a while to figure out the source of this issue.
In case this is not a bug, I think it would be helpful to add some information to the docs and/or change the error message

**Comment by ChrisRackauckas** [comment]
It needs a better error message.

**Comment by lungd** [comment]
Al right.
Feel free to close this issue unless you want to keep it as a reminder.

**Comment by ChrisRackauckas** [comment]
Let's keep it open until we get a better error message.

**Comment by contradict** [comment]
I think this was fixed by #649, the error message is now:

```
ERROR: ArgumentError: Value b provided in map not found in Sym{ModelingToolkit.Parameter{Real}}[a]
```

**Comment by ChrisRackauckas** [comment]
Thanks
